File: 10. Summation of primes/Javascript/solution.js
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

// Find the sum of all the primes below two million.

  var sumOfAllPrimes = function(n){
    var sum = 0;
    for(var i = 2; i <= n; i+=){
      if(isPrime(i)){
        sum += i
      }
    }
    return sum;
  };

  var isPrime = function(n){
    for(var i = 2; i <= Math.sqrt(n); i++){
      if (n % i === 0){
        return false;
      }
    }
    return true;
  };

File: 10. Summation of primes/Ruby/solution.rb
# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

# Find the sum of all the primes below two million.

  def sum_of_all_primes(n)
    sum = 0
    (1..n).each do |x|
      if prime? x
        sum += x
      end 
    end 
    sum
  end

  def prime?(n)
    (2..Math.sqrt(n)).each do |x|
      if n % x == 0
        return false
      end
    end
    true
  end